sweepbatcher/sweep_batch.go

```diff
@@ -1318,11 +1336,50 @@ func constructUnsignedTx(sweeps []sweep, address btcutil.Address,
 	fee := clampBatchFee(feeForWeight, batchAmt)
 
 	// Add the batch transaction output, which excludes the fees paid to
-	// miners.
-	batchTx.AddTxOut(&wire.TxOut{
-		PkScript: batchPkScript,
-		Value:    int64(batchAmt - fee),
-	})
+	// miners. Reduce the amount by the sum of change outputs, if any.
+	if len(changeOutputs) == 0 {
+		batchTx.AddTxOut(&wire.TxOut{
+			PkScript: batchPkScript,
+			Value:    int64(batchAmt - fee),
+		})
+	} else {
+		// Reduce the batch output by the sum of change outputs.
+		var sumChange int64
+		for _, change := range changeOutputs {
+			sumChange += change.Value
+		}
+		// Add the main output first.
+		batchTx.AddTxOut(&wire.TxOut{
+			PkScript: batchPkScript,
+			Value:    int64(batchAmt-fee) - sumChange,
+		})
+		// Then add change outputs.
+		for _, txOut := range changeOutputs {
+			batchTx.AddTxOut(&wire.TxOut{
+				PkScript: txOut.PkScript,
+				Value:    txOut.Value,
+			})
+		}
+	}
+
+	// Check that for each swap, inputs exceed the change outputs.
```

Comment:
This check is only needed when there is change.
```go
if len(changeOutputs) != 0 {
```
We can save a couple of `map` allocations with this `if`.